src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/util/DynamicFrameworkContractFactory.kt

```diff
@@ -27,6 +27,7 @@ class DynamicFrameworkContractFactory(em: TestEntityManager? = null) : EntityFac
     npsRegion: NPSRegion? = null,
     pccRegion: PCCRegion? = null,
     contractReference: String = RandomStringUtils.randomAlphanumeric(8),
+    subcontractor: Set<ServiceProvider>? = null,
```

Comment:
nit: most of these classes set the default params in the signature, instead of lower down. this means the type doesn't have to be nullable. e.g. contractReference above.